skbeam/core/accumulators/bin_image.py

```diff
@@ -0,0 +1,219 @@
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import numpy as np
+
+
+class BinImage(object):
+    """
+    Class to bin an ndarray onto an axis using a
+    user-specified bin for each element of the array.
+    """
+
+    def __init__(self, bin_values, norm, nbins, min=None, max=None,
+                 weights=None):
+        """
+        Parameters
+        ----------
+        bin_values : np.ndarray
+            For each pixel, this is the value that determines which bin
+            a pixel's intensity will contribute to.  It can be an arbitrary
+            array of floats/ints which will be discretized into nbins equally
```

Comment:
I mean that is up to you. One could take your current class and add the functionality to it. ie. if someone hands it a statistic function or their own function then use a different chunk of code. Incidentally, I have written a prototype of binned_statistic which uses multiprocessing, and does all the statistical measures in one shot. https://gist.github.com/CJ-Wright/747476a33ada014d7d91f6731a4c77f3 I don't know if that is helpful or not. You should also fee free to ignore me, I mean it sounds like these might be different use cases and one does not invalidate the other.